internal/services/automation/automation_powershell72_module_resource.go

```diff
@@ -0,0 +1,374 @@
+					"uri": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+
+					"hash": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"algorithm": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+								},
+								"value": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r AutomationPowerShell72ModuleResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r AutomationPowerShell72ModuleResource) ModelObject() interface{} {
+	return &AutomationPowerShell72ModuleModel{}
+}
+
+func (r AutomationPowerShell72ModuleResource) ResourceType() string {
+	return "azurerm_automation_powershell72_module"
+}
+
+func (r AutomationPowerShell72ModuleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return module.ValidatePowerShell72ModuleID
+}
+
+func (r AutomationPowerShell72ModuleResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Automation.Module
+
+			var model AutomationPowerShell72ModuleModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			name := metadata.ResourceData.Get("name").(string)
+
+			id := module.NewPowerShell72ModuleID(subscriptionId, model.ResourceGroupName, model.AutomationAccountName, name)
+
+			existing, err := client.PowerShell72ModuleGet(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			// for existing global module do update instead of raising ImportAsExistsError
+			isGlobal := existing.Model != nil && existing.Model.Properties != nil && existing.Model.Properties.IsGlobal != nil && *existing.Model.Properties.IsGlobal
+			if !response.WasNotFound(existing.HttpResponse) && !isGlobal {
+				return tf.ImportAsExistsError("azurerm_automation_powershell72_module", id.ID())
+			}
+
+			parameters := module.ModuleCreateOrUpdateParameters{
+				Properties: module.ModuleCreateOrUpdateProperties{
+					ContentLink: expandPowerShell72ModuleLink(model.ModuleLink),
+				},
+			}
+
+			if _, err := client.PowerShell72ModuleCreateOrUpdate(ctx, id, parameters); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			deadline, ok := ctx.Deadline()
+			if !ok {
+				return fmt.Errorf("internal-error: context had no deadline")
+			}
+
+			// the API returns 'done' but it's not actually finished provisioning yet
```

Comment:
We have an open PR on the MSFT rest api specs repo to fix this issue, would you mind updating the comment to link that here:
https://github.com/Azure/azure-rest-api-specs/pull/25435

Reply:
Good catch! Do you want me to fix it via https://github.com/hashicorp/pandora/blob/main/tools/importer-rest-api-specs/components/parser/dataworkarounds/workaround_automation_25435.go as an LRO for an even more thorough fix?

Reply:
You could, but it would mean you'd also need to add all of the polling statuses to the `go-azure-sdk` as well.
https://github.com/hashicorp/go-azure-sdk/blob/90c224fb9db7ea1b0e34036896c1f11ef57994ce/sdk/client/resourcemanager/poller_lro.go#L157

Since this is being tracked in an issue It's also fine to leave this as is if it's working as long as the comment is updated to link it to the issue.

Reply:
Done ✅